lyncs_io/numpy.py

```diff
@@ -14,20 +14,63 @@
 ]
 
 from io import UnsupportedOperation
+from functools import wraps
 import numpy
 from numpy.lib.npyio import NpzFile
-from numpy.lib.format import read_magic, _check_version, _read_array_header
+from numpy.lib.format import (
+    read_magic,
+    _check_version,
+    _read_array_header,
+    _write_array_header,
+    header_data_from_array_1_0,
+)
 from lyncs_utils import is_keyword
 from .archive import split_filename, Data, Loader, Archive
 from .header import Header
 from .utils import swap, open_file
+from .mpi_io import MpiIO
 
-load = numpy.load
-save = swap(numpy.save)
 loadtxt = numpy.loadtxt
 savetxt = swap(numpy.savetxt)
 
 
+@wraps(numpy.load)
+def load(filename, chunks=None, comm=None, **kwargs):
+    """
+    chunks = number of chunks per dir
+    comm = cartesian MPI_Comm
+    """
+
+    if comm is None or comm.size == 1:
+        return numpy.load(filename, **kwargs)
+
+    metadata = head(filename)
+
+    with MpiIO(comm, filename, mode="r") as mpiio:
+        local_array = mpiio.load(
+            metadata["shape"], metadata["dtype"], "C", metadata["_offset"]
+        )
+
+    return local_array
+
+
+@wraps(numpy.save)
+def save(array, filename, comm=None, **kwargs):
+
+    if comm is None or comm.size == 1:
+        return numpy.save(filename, array, **kwargs)
+
+    with MpiIO(comm, filename, mode="w") as mpiio:
+        global_shape, _, _ = mpiio.decomposition.compose(array.shape)
```

Comment:
Can this be moved in save?
i.e. have here only `mpiio.save(array)`

Reply:
Actually not, is fine here because the numpy header is specific of numpy. This it ok